user/using-debug-vm.md

```diff
@@ -0,0 +1,168 @@
+---
+title: Using Debug VM
+layout: en
+permalink: /user/using-debug-vm/
+---
+
+When the build failure is only observable on Travis CI, and if you suspect that there are
+critical differences between your local development environment and Travis CI's build VM,
+you can restart builds in the debug mode to gain access to the actual VM or container
+(henceforth "debug VM").
+
+# Restarting a build to debug
+
+This feature is available for private repositories and those public repositories for which
+the feature is enabled.
+
+In addition, the push access to the repository is required.
+
+For private repositories, the "Debug build" or "Debug job" button is available on the upper right corner of
+the build/job page.
+
+![Screenshot of debug build/job buttons](/images/debug_buttons.png)
+
+For public repositories, an API call is required, in addition to the feature being enabled.
+
+## Starting a debug VM via API
+
+In addition to using the debug button, a debug build may be started via
+API (which is the only way to do it for public repositories).
+
+Here is an example for a public repository.
+Token should be set to your Travis CI token, and the job ID (`$id`) should point to the job you
+want to debug.
+
+```sh-session
+$ curl -s -X POST \
+  -H "Content-Type: application/json" \
+  -H "Accept: application/json" \
+  -H "Travis-API-Version: 3" \
+  -H "Authorization: token ********************" \
+  -d "{\"quiet\": true}" \
+  https://api.travis-ci.org/job/${id}/debug
+```
+
+For private repositories, the API endpoint is `https://api.travis-ci.com/job/${id}/debug`.
+
+The JSON payload could be empty (`-d "{}"`), in which case any command input and output
+you see while connected to the debug VM will be reflected on the build log.
+
+# Connecting to the debug VM
+
+Once the debug VM starts, it will go through the intial steps, including setting up language
+run times and installing debug utilities.
+
+When the debug VM is ready for your connection, you will see:
+
+```
+Debug build initiated by BanzaiMan
+Setting up debug tools.
+Preparing debug sessions.
+Use the following SSH command to access the interactive debugging environment:
+ssh DwBhYvwgoBQ2dr7iQ5ZH34wGt@ny2.tmate.io
+This build is running in quiet mode. No session output will be displayed.
+This debug build will stay alive for 30 minutes.
+```
+
+Running the `ssh` command above will drop you in on a live VM.
+
+## Security implications of the `ssh` command
```

Comment:
That would imply that there is something inherently insecure about the debug build itself, which is not true.

Maybe: Security implications of public display of the `ssh` command